### PR TITLE
Add setting to control what happens when double-clicking RAM watch 

### LINKED ISSUE #3688: [Feature Request] Double click RAM watch to poke
Personally I tend to poke watches far more often than I edit their properties, so I'd like to open the poke window with a double click (instead of opening the edit window as it currently does). 

I can create a PR for this but wanted to ask for feedback first. I figure the way to do this would be either

1. Add a setting that changes what the double click does to the ram watch menu
2. Open poke when double clicking the value cell, open edit on the other cells
I prefer this because it's more flexible, but I don't know how intuitive this is to other users
3. Both (add settings for always edit/always poke/depending on cell)
4. Always poke

### PR DESCRIPTION
**Pull request by kalimag**
- Add a new setting to the `RamWatch` dialog menu that changes whether double-clicking a watch in the list will open the Edit or the Poke dialog. 
- Setting defaults to poke (previous behavior was edit)
- Pressing enter also obeys this setting 

Resolves #3688

Check if completed:
- [ ] I have run any relevant test suites
- [x] I, the committer, have read the [licensing terms for contributors](https://github.com/TASEmulators/BizHawk/blob/master/contributing.md#copyrights-and-licensing) (last updated 2022-07-15) and am compliant

### CODE REVIEW

File: src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.Designer.cs [hunk 1]
```diff
@@ -659,6 +663,24 @@ private void InitializeComponent()
             this.WatchListView.KeyDown += new System.Windows.Forms.KeyEventHandler(this.WatchListView_KeyDown);
             this.WatchListView.MouseDoubleClick += new System.Windows.Forms.MouseEventHandler(this.WatchListView_MouseDoubleClick);
             // 
+            // DoubleClickActionSubMenu
+            // 
+            this.DoubleClickActionSubMenu.DropDownItems.AddRange(new System.Windows.Forms.ToolStripItem[] {
+            this.DoubleClickToEditMenuItem,
+            this.DoubleClickToPokeMenuItem});
+            this.DoubleClickActionSubMenu.Text = "Double-Click Action";
```
Comment (YoshiRulz) [inline src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.Designer.cs:671]:
```suggestion
            this.DoubleClickActionSubMenu.Text = "On Double-Clicking a Watch:";
```
Though if Enter has the same effect, I'm not sure...

File: src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.cs [hunk 2]
```diff
@@ -494,6 +507,22 @@ private void EditWatch(bool duplicate = false)
 			}
 		}
 
+		private void PokeAddress()
```
Comment (YoshiRulz) [inline src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.cs:510]: I think the diff would be smaller if this was beside `PokeAddressMenuItem_Click`.

File: src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.cs [hunk 3]
```diff
@@ -433,6 +435,17 @@ private void FullyUpdateWatchList()
...
 
+		/// <summary>
+		/// Open Edit or Poke window for selected watch depending on user setting
+		/// </summary>
+		private void OpenWatch()
+		{
+			if (Settings.DoubleClickToPoke)
+				PokeAddress();
+			else
+				EditWatch();
```
Comment (YoshiRulz) [inline src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.cs:446]:
```diff
- if (Settings.DoubleClickToPoke)
- 	PokeAddress();
- else
- 	EditWatch();
+ if (Settings.DoubleClickToPoke) PokeAddress();
+ else EditWatch();
```
(or use braces)

**Review by YoshiRulz — CHANGES_REQUESTED** [review]

File: src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.Designer.cs (hunk 1 above)
Comment (kalimag) [reply src/BizHawk.Client.EmuHawk/tools/Watch/RamWatch.Designer.cs:671]: I figured keyboard navigation should also have access to that feature, but that most users wouldn't care or would figure it out. Explicitly referring to double-clicking seemed more intuitive overall than trying to find an all-encompassing term, and it didn't seem worth it to over-complicate the menu or have a super long description somewhere

**Review by YoshiRulz — APPROVED** [review]
LGTM now.

**Pull request merged**